src/ui/ResultsPreferences/ResultsPreferences.ts

```diff
@@ -110,70 +126,112 @@ export class ResultsPreferences extends Component {
+          this.save();
+          this.queryController.executeQuery({
+            closeModalBox: false
+          });
+        }, label, 'coveo-results-preferences-query-syntax');
+        this.preferencePanelRadioInputs[label] = radio;
+        return radio;
+      });
+      const formGroup = new FormGroup(radios, l('EnableQuerySyntax'));
+      $$(this.element).append(formGroup.build());
+      radios[2].select();
+      this.fromPreferencesToRadioInput();
+    }
   }
 
   private buildCheckboxesInput() {
-    var inputs = [];
+
+    const createCheckbox = (label: string) => {
+      const checkbox = new Checkbox((checkboxInstance) => {
+        this.fromPreferenceChangeEventToUsageAnalyticsLog(checkboxInstance.isSelected() ? 'selected' : 'unselected', label);
+        this.save();
+        this.queryController.executeQuery({
+          closeModalBox: false
+        });
+      }, label);
+      this.preferencePanelCheckboxInputs[label] = checkbox;
+      return checkbox;
+    };
+
+    const checkboxes: Checkbox[] = [];
+
     if (this.options.enableOpenInOutlook) {
-      inputs.push({ label: l('OpenInOutlookWhenPossible') });
+      checkboxes.push(createCheckbox(l('OpenInOutlookWhenPossible')));
+
     }
     if (this.options.enableOpenInNewWindow) {
-      inputs.push({ label: l('AlwaysOpenInNewWindow') });
+      checkboxes.push(createCheckbox(l('AlwaysOpenInNewWindow')));
     }
-    this.preferencePanelCheckboxInput = new PreferencesPanelCheckboxInput(inputs, ResultsPreferences.ID);
-    var container = $$('div', {
-      className: 'coveo-choices-container'
-    });
 
-    container.el.appendChild(this.preferencePanelCheckboxInput.build());
-    var executeOnChange = container.findAll('input');
-    _.each(executeOnChange, (toExec) => {
-      $$(toExec).on('change', (e: Event) => {
-        this.fromPreferenceChangeEventToUsageAnalyticsLog(e);
-        this.save();
-        this.queryController.executeQuery();
-      });
-    });
-
-    this.element.appendChild(container.el);
+    this.element.appendChild(new FormGroup(checkboxes, l('ResultLink')).build());
     this.fromPreferencesToCheckboxInput();
   }
 
-  private fromCheckboxInputToPreferences() {
-    var selected = this.preferencePanelCheckboxInput.getSelecteds();
+  private fromInputToPreferences() {
     this.preferences = {
       openInOutlook: false,
-      alwaysOpenInNewWindow: false
+      alwaysOpenInNewWindow: false,
+      enableQuerySyntax: undefined
     };
-    if (_.contains(selected, l('OpenInOutlookWhenPossible'))) {
-      this.preferences.openInOutlook = true;
-    }
-    if (_.contains(selected, l('AlwaysOpenInNewWindow'))) {
-      this.preferences.alwaysOpenInNewWindow = true;
-    }
+
+    _.each(this.preferencePanelCheckboxInputs, (checkbox: Checkbox, label: string) => {
+      if (this.isSelected(l('OpenInOutlookWhenPossible'), label, checkbox)) {
```

Comment:
`OpenInOutlookWhenPossible`, `AlwaysOpenInNewWindow`, `On`, `Off`, `Automatic` come back often, they should be in constants.